src/TestQuaternion.java

```diff
@@ -0,0 +1,22 @@
+import junit.framework.*;
+
+public class TestQuaternion extends TestCase {
+
+    public TestQuaternion(String name) {
+        super(name);
+    }
+
+    public void testEquals() {
+        Quaternion a = new Quaternion(1,0,-1,5);
+
+        assertTrue(a.equals(new Quaternion(1,0,-1,5)));
```

Comment:
That was one of the questions I had, because I tried with `assertEquals(a,new Quaternion(1,0,-1,5));` and got a message saying:
```
There was 1 failure:
1) testEquals(TestQuaternion)junit.framework.AssertionFailedError: expected:<( 1.0 ; 0.0 ; -1.0 ; 5.0 )> but was:<( 1.0 ; 0.0 ; -1.0 ; 5.0 )>
```

I did some research (http://stackoverflow.com/questions/6060848/junit-assertequals-fails-for-two-objects), and found that when you define a method called `equals()` for a class you create, you should write one with the following header, because this is the one JUnit uses in `assertEquals()`:

```
@Override
public boolean equals(Object other)
```

However, when I tried this I came across the problem that a generic Object doesn't have the fields (r,i,j,k) nor the setters and getters.

I searched how to define an `equals(Object)` method that overrides the default one and found (http://stackoverflow.com/questions/2316220/java-override-object-equals-method) that you can do something like check that the object is the type you want and then cast it, but I got a little confused by the comments, and because I was certain that `assertTrue()` worked, I used that.